pkg/provider-local/controller/service/add.go

```diff
@@ -71,17 +73,23 @@ func AddToManagerWithOptions(mgr manager.Manager, opts AddOptions) error {
 		return err
 	}
 
+	multiZone, err := hasNodesInMultipleZones(ctx, mgr.GetClient())
+	if err != nil {
+		return err
+	}
+
```

Comment:
You could use the "renowned" env var `SHOOT_FAILURE_TOLERANCE_TYPE` ([ref](https://github.com/gardener/gardener/blob/01b5894e86b97e6d17ec86e487b7fdb1d6870b87/test/e2e/gardener/e2e_suite_test.go#L41C48-L41C48)).

Reply:
That would indeed be a lot simpler, but unfortunately it is not set for the `provider-local` pod.